components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/UserNameMapperImpl.java

```
@@ -0,0 +1,113 @@
+
+/**
+ * This class is having implementation of UserNameMapping interface.
+ * Users can use instance of this class to get logged in user-id(whatever unique is to identify user) using provided
+ * pseudo name. Also to get logged in user's pseudo name(pseudo name used within APIM domain) using provided
+ * user id.
+ */
+public class UserNameMapperImpl implements org.wso2.carbon.apimgt.core.api.UserNameMapper {
+    private static final Logger log = LoggerFactory.getLogger(UserNameMapperImpl.class);
+    private UserMappingDAO userMappingDAO;
+
+    public UserNameMapperImpl() {
+        //
+    }
+
+    /**
+     * Initiate user name mapper with provided Data Access Object
+     *
+     * @param userMappingDAO Data access object to be used to retrieve user mapping.
+     */
+    public UserNameMapperImpl(UserMappingDAO userMappingDAO) {
+        this.userMappingDAO = userMappingDAO;
+    }
+
+    /**
+     * This method will return logged in user-id(whatever unique is to identify user) using provided
+     * pseudo name.
+     *
+     * @param pseudoName pseudo name of the user.
+     * @return user id of matching user in the system.
+     */
+    @Override
+    public String getLoggedInUserIDFromPseudoName(String pseudoName) {
+        //TODO implement method
+        //If pseudo name in map then get from that
+        //Else check mapping in database and load it to local map.
+        //If mapping is not in database then add it to db and cache both.
+        //then return name.
+        if (pseudoName != null && pseudoName.equalsIgnoreCase("admin")) {
+            return pseudoName;
+        } else {
+            try {
+                String userID = getUserMappingDAO().getUserIDByPseudoName(pseudoName);
+                return userID;
+            } catch (APIMgtDAOException e) {
+                log.error("Error while user getting user details for user : " + pseudoName);
+                return null;
```

Since this a clear error condition, lets throw an exception here instead of returning null and risking a NPE